homeassistant/components/rpi_gpio/light.py

```diff
@@ -0,0 +1,165 @@
+"""Support for controlling a Raspberry Pi light."""
+
+import time
+
+import voluptuous as vol
+
+from homeassistant.components import rpi_gpio
+from homeassistant.components.light import PLATFORM_SCHEMA, LightEntity
+from homeassistant.const import CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.reload import setup_reload_service
+
+from . import DOMAIN, PLATFORMS
+
+CONF_LIGHT = "lights"
+CONF_RELAY_PIN = "relay_pin"
+CONF_LIGHT_BUTTON_PIN = "light_button_pin"
+CONF_LIGHT_BUTTON_PULL_MODE = "light_button_pull_mode"
+CONF_INVERT_LIGHT_BUTTON = "invert_light_button"
+CONF_INVERT_RELAY = "invert_relay"
+CONF_LIGHT_BUTTON_BOUNCETIME_MILLIS = "light_button_bouncetime_millis"
+CONF_LIGHT_BUTTON_DOUBLE_CHECK_TIME_MILLIS = "light_button_double_check_time_millis"
+DEFAULT_LIGHT_BUTTON_PULL_MODE = "UP"
+DEFAULT_INVERT_LIGHT_BUTTON = False
+DEFAULT_INVERT_RELAY = False
+DEFAULT_LIGHT_BUTTON_BOUNCETIME_MILLIS = 150
+DEFAULT_LIGHT_DOUBLE_CHECK_TIME_MILLIS = 25
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept YAML configuration for platform configurations.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

These platforms need to be converted to an integration configuration first.

Reply:
do you have an example of a class with valid configuration please?
I took example on the class cover in the same package.

Reply:
@frenck Is this [documentation](https://developers.home-assistant.io/docs/creating_component_code_review/#2-configuration) referencing old structure? My impression is based on that decision the recommendation would be to use CONFIG_SCHEMA instead, is that right? (If so, I can send an update.)

Reply:
@frenck I did the modification, I understood how to do.
I have extracted light.py from rpi_gpio and created the package rpi_gpio_light.
So, for homogeneity, do you want I refactor the package rpi_gpio to extract cover, binary_sensor and switch into new packages to be compliant with ADR-0007?
I can do it in another pull request

Reply:
@allenporter is that modification what you where thinking?